tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/TestScenarioRunner.java

```diff
@@ -0,0 +1,230 @@
+                successfulTestCases++;
+            } catch (TestFailException ex) {
+                System.out.println("Test case failed: " + ex.toString());
+                System.out.println(ex.getMessage());
+                ex.printStackTrace();
+            }
+
+            testEnvironmentExecutor.close();
+        }
+
+        return successfulTestCases;
+    }
+
+    /**
+     * Run single test case
+     *
+     * @param testScenario Test scenario of the test case
+     * @param testCase Test case to run
+     * @param testEnvironmentExecutor Initialized environment
+     * @throws TestFailException
+     */
+    private void runTestCase(TestScenario testScenario, TestCase testCase,
+                             TestEnvironmentExecutor testEnvironmentExecutor) throws TestFailException {
+
+        System.out.println("Executing test case: " + testCase);
+
+        String environmentId = null;
+        try {
+            if (testEnvironmentExecutor instanceof DockerComposeExecutor) {
+                ((DockerComposeExecutor) testEnvironmentExecutor).createNetwork();
+            }
+
+            /**
+             * Start APM server
+             */
+            apmServer.setHost("0.0.0.0");
+            apmServer.setTraces(new ArrayList<>());
+            apmServer.run();
+
+            /**
+             * Run a container and wait
+             */
+            environmentId = testEnvironmentExecutor.run(testScenario.getEnvironment());
+            Thread.sleep(testScenario.getEnvironment().getInitWaitSeconds()*1000);
+
+            /**
+             * Execute test script and wait
+             */
+            testEnvironmentExecutor.execScript(environmentId, testCase.getScriptServiceName(), testCase.getAction());
+            Thread.sleep(testCase.getAfterActionWaitSeconds()*1000);
+
+            /**
+             * verify results
+             */
+            verifyResults(testScenario, testCase, apmServer);
+        } catch (InterruptedException ex) {
+            throw new TestFailException(testCase, ex);
+        } finally {
+            if (apmServer != null) {
+                apmServer.shutdown();
+            }
+
+            if (environmentId != null) {
+                testEnvironmentExecutor.clean(environmentId);
+            }
+        }
+    }
+
+    private void verifyResults(TestScenario testScenario, TestCase testCase, ApmMockServer apmServer) throws
+            TestFailException {
+
+        Collection<?> objects = getCapturedData(testScenario.getEnvironment().getType(), apmServer);
+
+        System.out.println("Captured objects:\n" + objects);
+
+        String json;
+        try {
+            json = serialize(objects);
+        } catch (IOException e) {
+            throw new TestFailException(testCase, "Failed to serialize traces", e);
+        }
+
+        for (JsonPathVerify jsonPathVerify: testCase.getVerify().getJsonPath()) {
+            if (!JsonPathVerifier.verify(json, jsonPathVerify)) {
```

Comment:
What if there are more than one path returning `false`? Wouldn't it be better to collect all failures and report them all, instead of failing on the first?